File: packages/create-react-native-library/templates/windows-cpp-library/windows/{%- project.name %}/ReactPackageProvider.cpp
#include "ReactPackageProvider.h"
#if __has_include("ReactPackageProvider.g.cpp")
#include "ReactPackageProvider.g.cpp"
#endif

#include <ModuleRegistration.h>

#include "<%- project.name %>_NativeModule.hpp"


namespace winrt::<%- project.name %>::implementation
{
    void ReactPackageProvider::CreatePackage( const winrt::Microsoft::ReactNative::IReactPackageBuilder& packageBuilder ) noexcept
    {
        winrt::Microsoft::ReactNative::AddAttributedModules( packageBuilder );
    }
}

File: packages/create-react-native-library/templates/windows-csharp-view-library/windows/{%- project.name %}/ReactPackageProvider.cs
using Microsoft.ReactNative;
using Microsoft.ReactNative.Managed;

namespace <%- project.name %>
{
    public partial class ReactPackageProvider : IReactPackageProvider
    {
        public void CreatePackage( IReactPackageBuilder packageBuilder )
        {
            packageBuilder.AddViewManagers();
            CreatePackageImplementation( packageBuilder );
        }

        /// <summary>
        /// This method is implemented by the C# code generator
        /// </summary>
        partial void CreatePackageImplementation( IReactPackageBuilder packageBuilder );
    }
}
